channeldb/payments.go

```diff
@@ -252,9 +252,71 @@ type Payment struct {
 	Failure *FailureReason
 }
 
+// ToMPPayment converts a legacy payment into an MPPayment.
+func (p *Payment) ToMPPayment() *MPPayment {
+	var (
+		htlcs   []HTLCAttempt
+		reason  *FailureReason
+		failure *HTLCFailInfo
+	)
+
+	// Promote the payment failure to a proper fail struct, if it exists.
+	if p.Failure != nil {
+		// NOTE: FailTime is not set for legacy payments.
+		failure = &HTLCFailInfo{}
+		reason = p.Failure
+	}
+
+	// Populate a single HTLC on the MPPayment if an attempt exists on the
+	// legacy payment. If none exists we will leave the attempt info blank
+	// since we cannot recover it.
+	if p.Attempt != nil {
+		// Promote the payment preimage to proper settle struct, if it
+		// exists.
+		var settle *HTLCSettleInfo
+		if p.Preimage != nil {
```

Comment:
I see no reason to have this inside the if though. Why not just handle it the same way as failure? (imagine sending a payment, but before you are able to make an attempt, somebody gives you the preimage and you can settle)

Reply:
reverted to prior structure